rigs/heft-web-rig/shared/webpack-base.config.js

```diff
@@ -0,0 +1,270 @@
+                    const useCssModules =
+                      // ...UNLESS the filename opts out using a file extension like "filename.global.scss"
+                      !/\.global\.\w+$/i.test(resourcePath) &&
+                      // ...UNLESS this is a .css file.
+                      !/\.css$/i.test(resourcePath);
+                    return useCssModules;
+                  },
+
+                  // This setting has no effect unless CSS modules is enabled. Possible values:
+                  // - "local": global CSS by default, overridable using the ":local" selector
+                  // - "global": local CSS by default, overridable using the ":global" selector
+                  // - "pure": requires selectors to contain at least one local class or id
+                  // - a lambda that returns the mode string; the function parameter is the resource path
+                  //
+                  // DEFAULT: "local"
+                  mode: 'local',
+
+                  // Set this to true if you want to be able to reference the global declarations using import statements
+                  // similar to local CSS modules
+                  //
+                  // DEFAULT: false
+                  // exportGlobals: true,
+
+                  // Provide a recognizable class/module names for developers
+                  //
+                  // DEFAULT: "[hash:base64]"
+                  localIdentName: production ? '[hash:base64]' : '[local]__[hash:base64:5]'
+                },
+
+                sourceMap: !production
+              }
+            }
+          ],
+
+          // Nested rules are applied after the parent rules.
+          // https://webpack.js.org/configuration/module/#nested-rules
+          rules: [
+            {
+              // Prevent postcss-loader and sass-loader from processing the .css file extension
+              test: /\.(scss|sass)$/,
+
+              use: [
+                {
+                  // PostCSS is a general-purpose CSS transformer; however, we prefer to avoid custom CSS syntaxes
+                  // and only use the standard SASS syntax.  Thus postcss-loader is used here only to apply the popular
+                  // "autoprefixer" plugin improves browser compatibility by generating vendor prefixes.
+                  // https://www.npmjs.com/package/postcss-loader
+                  loader: require.resolve('postcss-loader'),
+                  options: {
+                    postcssOptions: {
+                      plugins: [
+                        // https://www.npmjs.com/package/autoprefixer
+                        autoprefixer
+                      ]
+                    },
+
+                    sourceMap: !production
+                  }
+                },
+                {
+                  // Compiles SASS syntax into CSS
+                  // https://www.npmjs.com/package/sass-loader
+                  loader: require.resolve('sass-loader'),
```

Comment:
@iclanton is that possible today?

Reply:
Ian said that Heft has this feature today.  However, compiling `src/*.scss` -> `lib/*.css` today will conflict with this requirement:

https://github.com/microsoft/rushstack/blob/c4584201d42bf254469683cabb180142ea0c6c41/rigs/heft-web-rig/shared/webpack-base.config.js#L65-L72

To handle that, we need to add a `heft-sass-plugin` setting that allows the file extensions to be remapped (e.g. `src/*.scss` -> `lib/*.module.css`  BUT `src/*.global.scss` --> `lib/*.global.css`).

That needs to be implemented in a separate PR.

@dmichon-msft @jonasb @halfnibble

Reply:
When using `heft-sass-plugin` to compile SASS to CSS, the mapping is:
- `src/**/*.scss` -> `lib/**/*.scss.css`
- `src/**/*.sass` -> `lib/**/*.sass.css`
- `src/**/*.css` -> `lib/**/*.css`
This was done so that source code imports like:
```ts
import styles from './Foo.scss';
```
resolve correctly, whether you compile the file in advance or have webpack point at the source code. The only required change in webpack is to append `".css"` to `resolve.extensions`.

Reply:
@dmichon-msft I tried to add this, however it turns out that `sass.json` is not riggable because `"cssOutputFolders"` gets resolved relative to the rig project rather than the app project.  This is a design problem with `heft-sass-plugin`.

We should do a separate PR to fix `heft-sass-plugin`, then we can follow up with `heft-web-rig` to enable this optimization.